### PR TITLE
Fixed config locations to be URIs

### LINKED ISSUE #22699: 2.6 - Error with config locations
### Describe the bug

I have an `application.yaml` that gets used for all projects (containing things like `ignored-split-packages`). With the `ConfigGenerationBuildStep` changes in 2.6 I now get an exception during builds because of the following lines in my `build.gradle` file.

```groovy
    build {
        // Shared Quarkus configuration applied to all projects
        def shareConfig = rootProject.projectDir.toURI().toString() + "/conf/quarkus"
        System.setProperty("quarkus.config.locations", "${shareConfig}")
    }
```

It looks like `ConfigGenerationBuildStep` doesn't seem to like the URI string format (see exception below).

 I also tried  `toAbsolutePath()` instead of `toURI().toString()`, but then it pushed the error to `bstractLocationConfigSourceLoader`, which didn't like the drive letter when in Windows (see exception below).

### Expected behavior

Successful build

### Actual behavior

Error using `toURI().toString()`
```
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 4: file:/C:/my/git//conf/quarkus
        at io.quarkus.deployment.steps.ConfigGenerationBuildStep.lambda$watchConfigFiles$0(ConfigGenerationBuildStep.java:259)
        at io.quarkus.deployment.steps.ConfigGenerationBuildStep.watchConfigFiles(ConfigGenerationBuildStep.java:257)
```

Error using `toAbsolutePath()`
```
Caused by: java.lang.IllegalArgumentException: SRCFG00033: Scheme C not supported
        at io.smallrye.config.AbstractLocationConfigSourceLoader.loadConfigSources(AbstractLocationConfigSourceLoader.java:101)
        at io.smallrye.config.AbstractLocationConfigSourceLoader.loadConfigSources(AbstractLocationConfigSourceLoader.java:80)
        at io.smallrye.config.AbstractLocationConfigSourceFactory.getConfigSources(AbstractLocationConfigSourceFactory.java:30)
        at io.smallrye.config.PropertiesLocationConfigSourceFactory.getConfigSources(PropertiesLocationConfigSourceFactory.java:16)
        at io.smallrye.config.ConfigurableConfigSource.unwrap(ConfigurableConfigSource.java:53)
        at io.smallrye.config.ConfigurableConfigSource.getConfigSources(ConfigurableConfigSource.java:49)
        at io.smallrye.config.SmallRyeConfig$ConfigSources.mapLateSources(SmallRyeConfig.java:618)
        at io.smallrye.config.SmallRyeConfig$ConfigSources.<init>(SmallRyeConfig.java:529)
        at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:66)
        at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:419)
        at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:173)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:108)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:328)
```

### How to Reproduce?

```groovy
    build {
        // Shared Quarkus configuration applied to all projects
        def shareConfig = rootProject.projectDir.toURI().toString() + "/conf/quarkus"
        System.setProperty("quarkus.config.locations", "${shareConfig}")
    }
```

or

```groovy
    build {
        // Shared Quarkus configuration applied to all projects
        def shareConfig = rootProject.projectDir.getAbsolutePath() + "/conf/quarkus"
        System.setProperty("quarkus.config.locations", "${shareConfig}")
    }
```

### Output of `uname -a` or `ver`

MINGW64_NT-10.0-19042

### Output of `java -version`

OpenJDK 11.0.11

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.3.3

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mikethecalamity**
The [quarkus.config.locations](https://quarkus.io/guides/all-config#quarkus-core_quarkus.config.locations) configuration is specified as a `java.net.URI` in the documentation, but `ConfigGenerationBuildStep` was treating it as a file path

Fixes: https://github.com/quarkusio/quarkus/issues/22699

@radcortez @glefloch @gsmet

### CODE REVIEW

**Comment by mweber03** [comment]
Can this be tagged to main and 2.6.2?

**Review by radcortez — APPROVED** [review]

**Comment by radcortez** [comment]
Thanks for the PR :)

> Can this be tagged to main and 2.6.2?

It will be added automatically once merged.

**Comment by radcortez** [comment]
Ops, seems that the CI is not happy with the code style. Please do a maven build so the correct style is applied. Thank you!

**Comment by gsmet** [comment]
I'll fix the formatting issue but... I might be missing something.

We have in `ConfigConfig` (note the `String`):
```
    @ConfigItem(name = "config.locations")
    public Optional<List<String>> locations;
```
which is supposed to be a relocation for `smallrye.config.locations` so I'm unsure if reading this config property as URI would work? Or should it be URI everywhere?

Also the code has not been tested because it didn't compile (missing import and missing `;`) - I fixed it too - so I'm totally unsure it fixes the reported issue?

**Comment by radcortez** [comment]
> I'll fix the formatting issue but... I might be missing something.
> 
> We have in `ConfigConfig` (note the `String`):
> 
> ```
>     @ConfigItem(name = "config.locations")
>     public Optional<List<String>> locations;
> ```
> 
> which is supposed to be a relocation for `smallrye.config.locations` so I'm unsure if reading this config property as URI would work? Or should it be URI everywhere?

Actually, we only have `ConfigConfig` for documentation purposes. We don't use it, because this creates a chicken/egg problem since these configurations are used to configure the `Config` instance. We could update it to use `URI` anyway so it is correctly documented. 
 
> 
> Also the code has not been tested because it didn't compile (missing import and missing `;`) - I fixed it too - so I'm totally unsure it fixes the reported issue?

I was writing the exact same fix and I did test it in my Windows box.

**Comment by gsmet** [comment]
CI doesn't look happy. @radcortez I'll let you have a look and check if this is fixable or not.

**Comment by radcortez** [comment]
Ok, let me have a look.

**Comment by mweber03** [comment]
@radcortez Thanks for fixing it.

**Comment by gsmet** [comment]
@radcortez I would prefer if we adjusted `quarkus.config.locations` to be a list of URI, even if it's not strictly necessary.

**Comment by radcortez** [comment]
> @radcortez I would prefer if we adjusted `quarkus.config.locations` to be a list of URI, even if it's not strictly necessary.

Sure.

**Review by gsmet — APPROVED** [review]

**Pull request merged**
